tests/matrix_free/compute_diagonal_01.cc

```diff
@@ -25,7 +25,7 @@ template <int dim,
           typename Number              = double,
           typename VectorizedArrayType = VectorizedArray<Number>>
 void
-test()
+test(bool use_categorization = false)
```

Comment:
Could you add to the top of the file a description what the extended test is testing?